### ISSUE TITLE
V0.20.1 no longer supports/recognizes symlinks

### ISSUE THREAD

**Issue by inpharmaticist** [bitcoin/bitcoin#19899]
I have been keeping only a few files from the "blocks" folder on my local drive. The rest are moved to an external HDD but linked via symlinks. I upgrade from 0.19 and my symlinks aren't recognized, I keep getting "Error loading block database." If I move all files to my local machine, bitcoin core runs normally.

![image](https://user-images.githubusercontent.com/49789899/92331846-4d0bc400-f02e-11ea-8d52-1eac378e891c.png)

I am running Linux Mint and upgraded using snap (although I don't know what snap is, it seems like a new way to install Bitcoin Core on Mint).

**Comment by ArnaubHie** [comment]
Hi

Le dim. 6 sept. 2020 à 19:54, inpharmaticist <notifications@github.com> a
écrit :

**Comment by n-thumann** [comment]
I tried to replicate your setup (under macOS) by deleting most of the local data in `blocks` and creating identically named symlinks to my external drive and was able to reproduce your issue: Not the blockdata (`blk` and `rev` files), but the LevelDB Index folder is quite unhappy with symlinks. As soon a I symlinked the single files in `index` folder, I wasn´t able to run bitcoind and received the same error like you.
Solution: Don´t symlink the files inside `index`, but symlink the whole folder instead (`ln -s /path/to/blocks/index index` inside of your `blocks` folder) ✌️

**Comment by inpharmaticist** [comment]
Thanks for your response. That is interesting. I never moved anything from the index folder. I was only migrating the blk and rev files, and kept the two most recent on the local drive. I started doing this a few versions ago because I noticed startup was faster if at least some files were on the local drive. I tried symlinking the entire blocks folder with no improvement.

Then I tried using the HDD for the entire data directory and core won't open. 

![image](https://user-images.githubusercontent.com/49789899/92397481-ba296300-f0db-11ea-838b-e3ad1f3360e6.png)

Might go back to v0.19...
